ikarus/io/vtkwriter.hh

```diff
@@ -0,0 +1,182 @@
+// SPDX-FileCopyrightText: 2021-2024 The Ikarus Developers mueller@ibb.uni-stuttgart.de
+// SPDX-License-Identifier: LGPL-3.0-or-later
+
+/**
+ * \file vtkwriter.hh
+ * \brief Ikarus VTK Writer for finite element results
+ * \ingroup io
+ *
+ */
+
+#pragma once
+
+#include <dune/functions/gridfunctions/discreteglobalbasisfunction.hh>
+#include <dune/vtk/vtkwriter.hh>
+#include <dune/vtk/writers/unstructuredgridwriter.hh>
+
+#include <ikarus/io/resultfunction.hh>
+#include <ikarus/utils/concepts.hh>
+
+namespace Ikarus::Concepts {
+
+// adapted from /dune/dune-vtk/dune/vtk/utility/concepts.hh
+template <class DC>
+concept IsDataCollector = requires(DC dc) {
```

Comment:
I always wonder if we should put these concepts to the class itself or at to `ikarus/utils/concepts.hh` for now I would add this to the file `concepts.h` but we can discuss this seperately.